addons/mail/static/src/components/thread_viewer/thread_viewer_tests.js

```diff
@@ -476,6 +476,132 @@ QUnit.test('show message subject if thread is mailing channel', async function (
     )
 });
 
+QUnit.test('new messages separator', async function (assert) {
+    assert.expect(8);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                is_pinned: true,
+                message_unread_counter: 0,
+                name: "General",
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            session: {
+                name: 'Admin',
+                partner_display_name: 'Your Company, Admin',
+                partner_id: 3,
+                uid: 2,
+            },
+        },
+    });
+    const thread = this.env.models['mail.thread'].find(t => t.id === 20);
+    const threadViewer = this.env.models['mail.thread_viewer'].create({ thread: [['link', thread]] });
+    await this.createThreadViewerComponent(threadViewer, { hasComposer: true });
+
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_message',
+        "should have no messages"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_separatorNewMessages',
+        "should not display 'new messages' separator"
+    );
+
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "hey !"));
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    await afterNextRender(() => {
+        const messageData = {
+            type: 'author',
+            message: {
+                author_id: [this.env.session.partner_id, "Me"],
+                body: "hey !",
+                channel_ids: [20],
+                id: 2,
+                model: 'mail.channel',
+                res_id: 20,
+            },
+        };
+        const seenData = {
+            id: 1,
+            info: 'channel_seen',
+            last_message_id: 2,
+            partner_id: this.env.session.partner_id,
+        };
+        const notifications = [
+            [
+                ['my-db', 'res.partner', this.env.session.partner_id],
+                messageData,
+            ],
+            [
+                ['my-db', 'mail.channel', 20],
+                seenData,
+            ]
+        ];
+        this.widget.call('bus_service', 'trigger', 'notification', notifications);
+    });
```

Comment:
Test itself should not simulate that. Makes more sense to let mock server do this.
Maybe you need @seb-odoo's PR on changes in mock server before being able to do that. If so, put a comment above with task-id so that we can remove this in the future.

Reply:
Depends, if it's the seen triggered by the current user through using the components then yes, but if it's simulating the seen from another user then no.

Reply:
Here it looks like the seen notification generated by current user, so should be in mock server.